src/frontend/src/helpers/Storage/LocalStorage.ts

```diff
@@ -323,66 +281,61 @@ class Store implements StorageInterface {
   //   return new Dexie.Promise(function (resolve, reject) {resolve(true);});
   // }
 
-  public async newFile(proj: string, 
-                       file: string, 
-                       contents: string, 
-                       encoding: string, 
-                       normalize: boolean, 
-                       online_checksum: string): Promise<string> {
-    // account for base64 encoding
-    let rmatch: RegExpMatchArray;
-    if (encoding == "url" &&
-        (rmatch = contents.match(/^data:([^;]*)?(?:;(?!base64)([^;]*))?(?:;(base64))?,(.*)/))) {
-      const mime = rmatch[1];
-      const b64 = rmatch[3];
-      if(b64 || mime=="base64") {
-        contents = window.atob(rmatch[4]);
-      }
-    }
-    const checksum = (typeof contents === "string" && md5(contents)) || online_checksum || "";
-    const key = [proj, file];
+  public async newFile(proj: ProjectID,
+                       filename: string,
+                       contents?: string): Promise<FileID> {
+    // // account for base64 encoding
+    // let rmatch: RegExpMatchArray;
+    // if (encoding == "url" &&
+    //     (rmatch = contents.match(/^data:([^;]*)?(?:;(?!base64)([^;]*))?(?:;(base64))?,(.*)/))) {
```

Comment:
It is done in the backend call as well, but we need it somewhere in the frontend code for when we're not connected to the websocket